pageserver/src/tenant.rs

```diff
@@ -2672,8 +2678,8 @@ pub mod harness {
                 (Some(LOCK.read().unwrap()), None)
             };
 
-            let repo_dir = PageServerConf::test_repo_dir(test_name);
-            let _ = fs::remove_dir_all(&repo_dir);
+            let temp_repo_dir = tempfile::tempdir()?;
```

Comment:
I think this will create the dir in `/tmp` on most platforms, some of which use tmpfs to back it.
Are we creating large repos in unit tests?
That would cause tests to consume more DRAM than before.

Reply:
>  Are we creating large repos in unit tests?

No, we barely have unit tests at all, to start with.

>  That would cause tests to consume more DRAM than before.

I'm ok with that behavior as a default.